cmd/admin-bucket-remote-set.go

```diff
@@ -0,0 +1,214 @@
+		message += " "
+		message += console.Colorize("ARN", r.RemoteARN)
+		return message
+	case "remove":
+		return console.Colorize("RemoteMessage", "Removed Remote target for `"+r.SourceBucket+"` bucket successfully.")
+	case "set":
+		return console.Colorize("RemoteMessage", "Remote ARN = `"+r.RemoteARN+"`.")
+	}
+	return ""
+}
+
+// JSON returns jsonified message
+func (r RemoteMessage) JSON() string {
+	r.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(r, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// fetchRemoteTarget - returns the dest bucket, dest endpoint, access and secret key
+func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin.BucketTarget, err error) {
+	args := cli.Args()
+	argCount := len(args)
+	if argCount < 2 {
+		return sourceBucket, bktTarget, fmt.Errorf("Missing Remote target configuration")
+	}
+	_, sourceBucket = url2Alias(args[0])
+	TargetURL := args[1]
+	api := cli.String("api")
+	if api != "" && !isValidAPI(api) { // Empty value set to default "S3v4".
+		fatalIf(errInvalidArgument().Trace(api),
+			"Unrecognized API signature. Valid options are `[S3v4, S3v2]`.")
+	}
+	path := cli.String("path-style")
+	if !isValidPath(path) {
+		fatalIf(errInvalidArgument().Trace(path),
+			"Unrecognized bucket path style. Valid options are `[on,off, auto]`.")
+	}
+	u, cerr := url.Parse(TargetURL)
+	if cerr != nil {
+		fatalIf(probe.NewError(cerr), "Malformed Remote target URL")
+	}
+	arnType := cli.String("type")
+	if !madmin.ArnType(arnType).IsValid() {
+		fatalIf(errInvalidArgument().Trace(arnType), "Invalid --type flag. Valid option is `[replica]`.")
+	}
+	secure := u.Scheme == "https"
+	accessKey := u.User.Username()
+	secretKey, _ := u.User.Password()
+	TargetBucket := strings.TrimPrefix(u.Path, slashSeperator)
+	TargetBucket = strings.TrimSuffix(TargetBucket, slashSeperator)
+
+	console.SetColor(cred, color.New(color.FgYellow, color.Italic))
+	creds := &auth.Credentials{AccessKey: accessKey, SecretKey: secretKey}
+	bktTarget = &madmin.BucketTarget{TargetBucket: TargetBucket, Secure: secure, Credentials: creds, Endpoint: u.Host, Path: path, API: api, Type: madmin.ArnType(arnType)}
+	return sourceBucket, bktTarget, nil
+}
+
+// mainAdminBucketRemoteSet is the handle for "mc admin bucket remote set" command.
+func mainAdminBucketRemoteSet(ctx *cli.Context) error {
+	checkAdminBucketRemoteSetSyntax(ctx)
+
+	console.SetColor("RemoteMessage", color.New(color.FgGreen))
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+	// Create a new MinIO Admin Client
+	client, cerr := newAdminClient(aliasedURL)
+	fatalIf(cerr, "Unable to initialize admin connection.")
+
+	sourceBucket, bktTarget, perr := fetchRemoteTarget(ctx)
+	fatalIf(probe.NewError(perr), "Unable to parse input arguments.")
+	var arn string
+	arn, perr = client.SetBucketTarget(globalContext, sourceBucket, bktTarget)
```

Comment:
in mc Go error is called `e` and probe.Error is called `err`  so

```suggestion
	arn, e := client.SetBucketTarget(globalContext, sourceBucket, bktTarget)
```